brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformer.java

```diff
@@ -67,7 +67,7 @@ public ToscaTypePlanTransformer() {
     }
 
     @Override
-    public void setManagementContext(ManagementContext managementContext) {
+    public synchronized void setManagementContext(ManagementContext managementContext) {
```

Comment:
Ah, I had meant to move the synchronization from the level of the `initialiseAlien` up, so as to synchronise on the public method, `setManagementContext`, but forgot to remove the synchronized block! I don't think there's a need for the class level synchronisation, do you? I thought it was better to have the lock around both the setting of the management context in line 80 and its modification within the `initialiseAlien` method (via `.getBrooklynProperties().put(TOSCA_ALIEN_PLATFORM, platform)`

How does this look:
```
    @Override
    public synchronized void setManagementContext(ManagementContext managementContext) {
        if (!isEnabled()) {
            if (hasLoggedDisabled.compareAndSet(false, true)) {
                log.info("Not loading brooklyn-tosca platform: feature disabled");
            }
            return;
        }
        if (this.mgmt != null && this.mgmt != managementContext) {
            throw new IllegalStateException("Cannot switch mgmt context");
        } else if (this.mgmt == null) {
            this.mgmt = managementContext;
            initialiseAlien();
        }
    }

    private void initialiseAlien() {
        try {
            platform = mgmt.getConfig().getConfig(TOSCA_ALIEN_PLATFORM);
            if (platform == null) {
                Alien4CloudToscaPlatform.grantAdminAuth();
                if (platformFactory==null) platformFactory = new AlienPlatformFactory.Default();
                platform = platformFactory.newPlatform(mgmt);
                ((LocalManagementContext) mgmt).getBrooklynProperties().put(TOSCA_ALIEN_PLATFORM, platform);
            }
            alienInitialised.set(true);
        } catch (Exception e) {
            throw Exceptions.propagate(e);
        }
    }
```